server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -680,27 +682,39 @@ public static class CheckpointState implements Writeable {
          */
         long globalCheckpoint;
         /**
-         * whether this shard is treated as in-sync and thus contributes to the global checkpoint calculation
+         * whether this shard is treated as in-sync and has localTranslog, it contributes to the global checkpoint calculation.
          */
         boolean inSync;
 
         /**
-         * whether this shard is tracked in the replication group, i.e., should receive document updates from the primary.
+         * whether this shard is tracked in the replication group and has localTranslog, i.e., should receive document updates
+         * from the primary. Tracked shards with localTranslog would have corresponding retention leases on the primary shard's
+         * {@link ReplicationTracker}.
          */
         boolean tracked;
 
-        public CheckpointState(long localCheckpoint, long globalCheckpoint, boolean inSync, boolean tracked) {
+        // TODO - Need to check if CheckpointState or ReplicationTracker needs a remote stored tracking variable. This can be helpful later.
+
+        /**
+         * Whether this shard has a local translog copy. This would be true iff it is primary shard or relocating target of primary shard
+         * in case of remote translog store. Otherwise, this will be always true.
+         */
+        boolean localTranslog;
+
```

Comment:
We can introduce a generic state rather than coupling it to a very specific use case

Reply:
Ack.

Reply:
Done @Bukhtawar